### ISSUE TITLE
the gb image on my personal network 

### ISSUE THREAD

**Issue by Songtingt** [jacobgil/pytorch-grad-cam#63]
Hi, I am trying to apply this model on my personal network.
My network has two inputs of different sizes, when I visualize the gb image of the big image, only an area of the image that is the same size as the smaller image will show the gradient.
I would like to know how to solve this problem, thank you very much.

 ```
def __call__(self, input_small, input_big, target_category=None):
        if self.cuda:
            input_small = input_small.cuda()
            input_big = input_big.cuda()
        input_small= input_small.requires_grad_(True)
        input_big= input_big.requires_grad_(True)
        cls_score, offsets = self.forward(input_small, input_big) 

        B, _, H, W = cls_score.shape
        cls_score = cls_score.reshape(B, -1)  # 1,5329
        if target_category == None:
            target_category = np.argmax(cls_score.cpu().data.numpy())
        one_hot = np.zeros((1, cls_score.size()[-1]), dtype=np.float32)
        one_hot[0][target_category] = 1
        one_hot = torch.from_numpy(one_hot).requires_grad_(True)
        if self.cuda:
            one_hot = one_hot.cuda()
        one_hot = torch.sum(one_hot * cls_score) 
        one_hot.backward(retain_graph=True)  
        output = input_big.grad.cpu().data.numpy()
        # output2 = input_small.grad.cpu().data.numpy() #(1, 512, 512)
        output = output[0, :, :, :]  #(1, 800, 800)
        #output2=output2[0, :, :, :]  #(1, 512, 512)
        return output #,output2
```

**Comment by jacobgil** [comment]
Sorry for the late response.
- What is the size of input_big.grad.cpu()?
- input_big.grad.cpu() returns 0's?
- Is there a chance the network uses only input_small ?

**Comment by Songtingt** [comment]
Thank you for your reply!
- the size of **input_big.grad.cpu()** is (1,1,800,800), the function can alse return the output2 which the size is (1,1,512,512), the gradient image of the **input-small** is normal. (output2 is commented in the code)
- No, most areas are 0, except for areas of the same size as input_sar where the value is non-0.
- No, the network must input two pictures and after some operations finally returns a classification score map.

Here are the two gradient images of the output **(1,800,800)** and output2 **(1,512,512)**.
 ![the gradient image of the input_big](https://raw.githubusercontent.com/Songtingt/Similarity/main/input_big_gb.png)
 ![the gradient image of the input_small](https://raw.githubusercontent.com/Songtingt/Similarity/main/input_small_gb.png)

**Comment by jacobgil** [comment]
Thanks. 
Trying to understand this. 

Is there a chance the network is cropping the larger image, or something like that, cuasing the gradeints to really be non-zero in that area?

As a sanity check,
It might be worth to replace the network with a simple Module like
input_small.mean() + input_large.mean()
And see if then the gradients are still like this.
